### PR TITLE
Making Narayana version configurable

### PR DESCRIPTION
**Pull request by ochaloup**
Changing pom.xml files to not have hardcoded version for Narayana plus
unifying the name of property that is used for setting the Narayana
dependency version to 'version.narayana'.

### CODE REVIEW

**Comment by zhfeng** [comment]
It needs to re-test after the https://github.com/jbosstm/narayana/pull/1002 is merged

**Comment by zhfeng** [comment]
re-test

**Comment by zhfeng** [comment]
It needs to install the narayana-osgi-parent pom also

**Comment by ochaloup** [comment]
@zhfeng: does it mean that it needs a fix in Narayana or there is need some configuration change in quickstarts?

**Comment by zhfeng** [comment]
yeah, I had fixed in https://github.com/jbosstm/narayana/pull/1003

**Comment by zhfeng** [comment]
reset it again

**Comment by zhfeng** [comment]
@tomjenkinson I'm happy with these changes and thanks @ochaloup for the contribution

File: jca-and-hibernate/pom.xml [hunk 1]
```diff
@@ -43,7 +43,7 @@
         <version.org.hibernate>4.2.7.SP1</version.org.hibernate>
         <version.org.hibernate.hibernate-validator>5.0.1.Final</version.org.hibernate.hibernate-validator>
         <version.org.jboss.jboss-common-core>2.2.17.GA</version.org.jboss.jboss-common-core>
-        <version.org.jboss.narayana>${project.version}</version.org.jboss.narayana>
+        <version.narayana>${project.version}</version.narayana>
```
Comment (zhfeng) [inline jca-and-hibernate/pom.xml]: I assume that it is should be "<version.narayana>5.3.3.Final-SNAPSHOT</version.narayana>

Comment (ochaloup) [inline jca-and-hibernate/pom.xml]: ah, for sure, thanks. fixed

**Pull request merged**